lib/vsc/administration/ldapsync.py

```diff
@@ -177,34 +177,44 @@ def sync_altered_groups(self, last, dry_run=True):
         }
 
         for group in changed_groups:
-            vo = False
-            try:
-                vo = mkVo(self.client.vo[group.vsc_id].get()[1])
-            except HTTPError as err:
-                # if a 404 occured, the group is not an VO, so we skip this. Otherwise something else went wrong.
-                if err.code != 404:
-                    raise
+            # General group attributes
             group_moderators = [str(m) for m in group.moderators]
             institute_name = str(group.institute['name'])
-            if not group_moderators:
-                group_moderators = [str(VSC_CONFIG.backup_group_mods[group.institute['name']])]
-                logging.info("Using backup moderator %s for group %s", group_moderators, group.vsc_id)
+
             ldap_attributes = {
                 'cn': str(group.vsc_id),
                 'institute': [institute_name],
                 'gidNumber': ["%d" % (group.vsc_id_number,)],
                 'moderator': group_moderators,
-                'memberUid': [str(a) for a in group.members],
                 'status': [str(group.status)],
             }
-            if vo:
+
+            # Only set memberUid if there are actually active members in the group
+            # LDAP addition of records with empty memberUid will fail
+            if group.members:
+                ldap_attributes['memberUid'] = [str(a) for a in group.members]
```

Comment:
I thought this was a required attribute. So it will fail if there are no members, no?

Reply:
https://ldapwiki.com/wiki/PosixGroup only `gidNumber` is required.

Reply:
What if the LDAP now has a list of people and the update does not contain `memberUid` because all members went inactive? Will this overwrite the existing `mebermUid` if the attribute is missing?

Reply:
In such a case, the `memberUid` attribute will indeed be deleted from the existing LDAP record if it is missing in the new record. That's the default behaviour of [ldap.modlist.modifyModlist](https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap-modlist.html#ldap.modlist.modifyModlist) with `ignore_oldexistent=0`.
I made a tiny change in the related comment removing the mention to active members. The group members retrieved from the account page can be either active or inactive.

Reply:
Can you add what you say here too as a comment? If no `memberUid` is given, it will result in the deletion of any existing already in the ldap?

Reply:
Sure, done in https://github.com/hpcugent/vsc-administration/pull/102/commits/a89cce3d2ed0d6c434175c94f087502e07947efa
Let me know if it is clear enough, I tried to be concise.